src/SymfonyCache/CacheInvalidation.php

```diff
@@ -11,35 +11,32 @@
 
 namespace FOS\HttpCache\SymfonyCache;
 
-use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\HttpKernel\HttpCache\StoreInterface;
 use Symfony\Component\HttpKernel\HttpKernelInterface;
 
-/**
- * Interface for a HttpCache that supports active cache invalidation.
- */
-interface CacheInvalidation extends HttpKernelInterface
-{
-    /**
-     * Forwards the Request to the backend and determines whether the response should be stored.
-     *
-     * This methods is triggered when the cache missed or a reload is required.
-     *
-     * This method is present on HttpCache but must be public to allow event listeners to do
-     * refresh operations.
-     *
-     * @param Request $request A Request instance
-     * @param bool    $catch   Whether to process exceptions
-     *
-     * @return Response A Response instance
-     */
-    public function fetch(Request $request, $catch = false);
+use function class_alias;
+use function class_exists;
+
+if (! interface_exists(CacheInvalidation::class) && 
+    class_exists('Symfony\Component\HttpKernel\Kernel') &&
```

Comment:
please use a `use` statement and the `::class` pseudoconstant for this as well. like with the interface, it works to use a ::class even for things that don't exist.